xla/service/dump.cc

```diff
@@ -892,7 +892,7 @@ std::string GetNonDefaultDebugOptions(const DebugOptions& debug_options) {
       // Field exists in debug_options but not defaults
       absl::StrAppend(&non_default_options, field->name(), ": ",
                       GetValueAsString(reflection, debug_options, field), "\n");
-    } else if (reflection->HasField(debug_options, field)) {
+    } else {
```

Comment:
Do you need to check `reflection->HasField` above, when you check:

```
    if (reflection->HasField(debug_options, field) &&
        !reflection->HasField(default_options, field)) {
```

It presumably is possible to call GetValueAsString on a field that isn't explicitly set, so perhaps you don't have to call `reflection->HasField` at all.

Reply:
That is a good point. Removed the check for `HasField`.